jenkins/test/testsuite/Jenkinsfile

```diff
@@ -68,20 +75,60 @@ pipeline {
             post() {
                 always {
                     script {
-                        publish()
+                        uploadArtifacts(
+                            upload: true,
+                            build_number: "${build_id}",
```

Comment:
Where's build_id coming from? Is it used? Should it be `$BUILD_ID`. 

Also `"${x}"` is the same as `$x`.

Reply:
This is the build_id on which we want to run tests on.

Reply:
Shouldn't it be `BUILD_ID` or `buildId`?

Reply:
If you say BUILD_ID thats jenkins build_id (run number) I think we are looking for opensearch/dashboards build-id (basically jenkins run number of bundle-build) so that we grab right artifacts to run test aganist

Reply:
I see. We should rename that. It's too confusing to have `BUILD_ID` and `build_id`, but I understand what's going on now.

Currently `uploadArtifacts` takes a manifest and uploads `builds` and `dist`.

I think it would be totally fine to have `uploadTestResults` if these two re sufficiently different (looks like we don't need to read a manifest for test results because we pass in architecture and platform into the actual test job?). Then we don't need to make it parameterized that much. You could maybe extract the part in `withAWS` into `uploadDirs` still.

Reply:
Yes, that's the case. We can name it as `build-id`.